eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/SubmittedVariantAccessioningService.java

```diff
@@ -17,18 +17,114 @@
 
-public class SubmittedVariantAccessioningService extends BasicMonotonicAccessioningService<ISubmittedVariant, String> {
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class SubmittedVariantAccessioningService implements AccessioningService<ISubmittedVariant, String, Long> {
+
+    private static final Long ACCESSIONING_MONOTONIC_INIT_SS = 5000000000L;
+
+    private BasicMonotonicAccessioningService<ISubmittedVariant, String> accessioningService;
+
+    private BasicMonotonicAccessioningService<ISubmittedVariant, String> accessioningServiceDbsnp;
 
     public SubmittedVariantAccessioningService(MonotonicAccessionGenerator<ISubmittedVariant> accessionGenerator,
-                                               MonotonicDatabaseService dbService) {
-        super(accessionGenerator, dbService, new SubmittedVariantSummaryFunction(), new SHA1HashingFunction());
+                                               MonotonicDatabaseService dbService,
+                                               MonotonicDatabaseService dbServiceDbsnp) {
+        this.accessioningService = new BasicMonotonicAccessioningService<ISubmittedVariant, String>
+                (accessionGenerator, dbService, new SubmittedVariantSummaryFunction(), new SHA1HashingFunction());
+        this.accessioningServiceDbsnp = new BasicMonotonicAccessioningService<ISubmittedVariant, String>
+                (accessionGenerator, dbServiceDbsnp, new SubmittedVariantSummaryFunction(), new SHA1HashingFunction());
     }
 
-}
+    @Override
+    public List<AccessionWrapper<ISubmittedVariant, String, Long>> getOrCreate(
+            List<? extends ISubmittedVariant> variants)
+            throws AccessionCouldNotBeGeneratedException {
+        List<AccessionWrapper<ISubmittedVariant, String, Long>> dbsnpVariants = accessioningServiceDbsnp.get(variants);
+        List<ISubmittedVariant> variantsNotInDbsnp = removeFromList(variants, dbsnpVariants);
+        List<AccessionWrapper<ISubmittedVariant, String, Long>> submittedVariants = accessioningService.getOrCreate(variantsNotInDbsnp);
+        return joinLists(submittedVariants, dbsnpVariants);
+    }
+
+    private List<ISubmittedVariant> removeFromList(List<? extends ISubmittedVariant> allVariants,
+                                                   List<AccessionWrapper<ISubmittedVariant, String, Long>>
+                                                           vatiantsToDelete) {
+        return allVariants.stream().filter(variant -> !contains(vatiantsToDelete, variant))
+                          .collect(Collectors.toList());
+    }
+
+    private Boolean contains(List<AccessionWrapper<ISubmittedVariant, String, Long>> accessionWrapperList,
+                             ISubmittedVariant iSubmittedVariant) {
+        return accessionWrapperList.stream().anyMatch(x -> x.getData().equals(iSubmittedVariant));
+    }
+
+    private List<AccessionWrapper<ISubmittedVariant, String, Long>> joinLists(List l1, List l2) {
```

Comment:
why not using generics here? if you only use this function for accession wrappers, use that type